### ISSUE TITLE
[nbox/n2disk] Interfaces with udev modified names not recognized by n2disk section of nBox

### ISSUE THREAD

**Issue by thecityofguanyu** [ntop/n2disk#5]
This is probably more of an nBox issue than an n2disk issue, but this was the most relevant place I could find to open an issue.

We rename interfaces via udev rules:
```
root@pcap00:/storage/n2disk/xmr-ptc-tap# cat /etc/udev/rules.d/70-persistent-net.rules | grep xmr
SUBSYSTEM=="net", ACTION=="add", DRIVERS=="?*", ATTR{address}=="a0:36:9f:8b:a0:dc", ATTR{dev_id}=="0x0", ATTR{type}=="1", NAME="xmr-ptc-tap"

root@pcap00:/storage/n2disk/xmr-ptc-tap# ifconfig xmr-ptc-tap
xmr-ptc-tap Link encap:Ethernet  HWaddr a0:36:9f:8b:a0:dc
          inet6 addr: fe80::a236:9fff:fe8b:a0dc/64 Scope:Link
          UP BROADCAST RUNNING PROMISC MULTICAST  MTU:9710  Metric:1
          RX packets:313948571 errors:0 dropped:0 overruns:0 frame:0
          TX packets:4 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:284011121856 (284.0 GB)  TX bytes:324 (324.0 B)
          Memory:91d00000-91e00000
```

The dashboard and PF_RING/ZC pages of nBox seem to detect the renamed interfaces fine:

![dash](http://i.imgur.com/7alruDZ.png)

![pfring](http://i.imgur.com/ORmUzsp.png)

The n2disk page, however, does not recognize it as an interface. The `enp5s0f*` interfaces were the names of the interfaces before they were renamed. The `xmr-ptc-tap` you see below is from trying to manually edit that non-existent interface via just replacing the bad ifname with that in the URL. That cause it to "create" in some capacity, as you can see; however, it still thinks that it is down and unsuable.

![n2disk page](http://i.imgur.com/qrABYkr.png)

![n2disk if page](http://i.imgur.com/qv8Jcf0.png)

n2disk works just fine with the renamed interface when invoked from CLI. It just seems to be an nBox thing -- at least as far as I can identify.

**Comment by ValentinaViscarelli** [comment]
I have made a modify. Now you should see the interface in n2disk configuration.
You will able to find the modify in tomorrow build (dev packages).
Please update and give it a try.

**Comment by thecityofguanyu** [comment]
Hi Valentina,

I updated to the latest available dev packages.

```
caevans@btctul00-pcap00:/ sudo dpkg -l | grep ntop | awk '{print $2,$3,$4}'
apt-ntop 2.5-505 all
e1000e-zc-dkms 3.2.7.1.1229 all
fm10k-zc-dkms 0.20.1.1229 all
igb-zc-dkms 5.3.3.5.1229 all
ixgbe-zc-dkms 4.1.5.1229 all
nbox 2.5-716 all
ntopng 2.5.170516-2838 amd64
ntopng-data 2.5.170516 all
pfring 6.7.0-1229 amd64
pfring-drivers-zc-dkms 1.3 all
```

The renamed interfaces now appear, but you cannot actually start a capture. An "updating" message stays in front of them and they are non-functional from the UI.

See the screenshot below.

![interface updating](http://i.imgur.com/c4QzY8O.png)

**Comment by ValentinaViscarelli** [comment]
Hi @thecityofguanyu 
The problem is that nbox doesn't manage characters like "-" or "_" in the interface names. You should use only letters/numbers. For example in your case you could call your interface "cerPtcTap".

**Comment by thecityofguanyu** [comment]
Hi @ValentinaViscarelli 

I removed the special characters as you suggested. The "updating" is now gone. 

For some reason, the `-i` parameter wasn't added to any of the `/etc/n2disk/n2disk-example.conf` files, so they weren't capturing the right interfaces. I removed all the `.conf` files and reconfigured the interfaces from the WebUI. They then created properly and I was able to capture from WebUI.

Thanks for all the help!
